### PR TITLE
Fix #3558: Merge profile_chooser_profile_view.xml into single xml file

### LINKED ISSUE #3558: Merge profile_chooser_profile_view.xml into single xml file
Currently there are 4 versions of `profile_chooser_profile_view.xml` file merge it into single xml file.

We can use https://text-compare.com/ to compare two versions of this file and for all the differences we can create variables in `dimens.xml` file and use it accordingly.

It can happen that it becomes really hard to merge all 4 files into single file, in such cases prefer code-readability over single file. Therefore it can be fine if in the end the code still contains 2 or 3 files.

**Note**: **In PR, make sure you add before and after screenshot of mobile-portrait, mobile-landscape, tablet-portrait and tablet-landscape for comparison and make sure that there is not difference between before and after UI.**

### PR DESCRIPTION
**Pull request by dmdbilal**
Fixes #3558

### CODE REVIEW

**Comment by yash10019coder** [comment]
Hi @dmdbilal also refernce the issue to this PR by add Fixes #3558 into this PR description

**Comment by dmdbilal** [comment]
> Hi @dmdbilal also refernce the issue to this PR by add Fixes #3558 into this PR description

@yash10019coder Referenced the issue in the PR description.

File: app/src/main/res/layout-land/profile_chooser_profile_view.xml [hunk 1]
```diff
@@ -94,4 +96,4 @@
       app:layout_constraintStart_toEndOf="@+id/profile_chooser_item"
       app:layout_constraintTop_toTopOf="parent" />
   </androidx.constraintlayout.widget.ConstraintLayout>
-</layout>
+</layout>
```
Comment (yash10019coder) [inline app/src/main/res/layout-land/profile_chooser_profile_view.xml]: add newline at the end.

File: app/src/main/res/layout/profile_chooser_profile_view.xml [hunk 2]
```diff
@@ -102,4 +108,4 @@
       app:layout_constraintStart_toStartOf="parent"
       app:layout_constraintTop_toBottomOf="@+id/profile_chooser_item" />
   </androidx.constraintlayout.widget.ConstraintLayout>
-</layout>
+</layout>
```
Comment (yash10019coder) [inline app/src/main/res/layout/profile_chooser_profile_view.xml]: same here

**Review by yash10019coder — CHANGES_REQUESTED** [review]
Hi @dmdbilal I recently tried to run the app from your latest code on the PR it seemed to work I tried on both tablet and mobile in both portrait and landscape. 

Now your can add the screenshots of the old and new UI in the PR description and also add the passing test cases for files ProfileChooserFragmentTest and ProfileChooserActivityTest

Thanks,
Yash

**Comment by yash10019coder** [comment]
also please resolve merge conflicts in your PR

**Comment by yash10019coder** [comment]
Here @dmdbilal https://www.jetbrains.com/help/idea/resolving-conflicts.html#distributed-version-control-systems this article will help you to resolve merge conflict.

**Comment by yash10019coder** [comment]
Hi @dmdbilal try viewing this video it may help
https://www.youtube.com/watch?v=4ag6lJaSwzc

**Comment by yash10019coder** [comment]
Hi @dmdbilal please try to setup a upstream according to this 
https://github.com/oppia/oppia-android/wiki/Fork-and-Clone-Oppia-Android#fork-and-clone-the-oppia-android-repository
after setting the upstream  run this command to get the latest develop code.
```
git pull upstream develop
```

**Comment by dmdbilal** [comment]
Ok I'll try these methods.

**Comment by dmdbilal** [comment]
@yash10019coder I've resolved the merge conflicts and the app works well in both mobile and tab. But the problem is when i run the ProfileChooserFragmentTest and ProfileChooserActivityTest I'm getting this error

```
Error running 'ProfileChooserActivityTest': 
Class 'org.oppia.android.app.profile.ProfileChooserActivityTest' not found in module 'oppia-android.app.unitTest'
```
```
Error running 'ProfileChooserFragmentTest': 
Class 'org.oppia.android.app.profile.ProfileChooserFragmentTest' not found in module 'oppia-android.app.unitTest'
```

**Comment by yash10019coder** [comment]
Try rebuilding the project after invalidating the caches

**Comment by dmdbilal** [comment]
> Try rebuilding the project after invalidating the caches

@yash10019coder I've rebuild and re-run the test now, but still it shows the same error.

**Comment by yash10019coder** [comment]
@dmdbilal you can follow this https://github.com/oppia/oppia-android/wiki/Oppia-Android-Testing#running-robolectric-tests

**Comment by dmdbilal** [comment]
> @dmdbilal you can follow this https://github.com/oppia/oppia-android/wiki/Oppia-Android-Testing#running-robolectric-tests

### Still I'm getting the same issue here.
Screenshot 2022-12-22 203126

**Comment by dmdbilal** [comment]
@yash10019coder I've rebuilt the project many times but still I'm getting this same issue again and again.

**Comment by yash10019coder** [comment]
Hi @rt4914 could you approve the workflow run for this PR.

**Comment by dmdbilal** [comment]
> workflow run

What's that workflow run ?

**Comment by yash10019coder** [comment]
@dmdbilal a workflow run is like running all the test cases and other ci/cd(continuous integration and deployment) on the GitHub to test for any errors.
Continuous integration is like for the development whenever a developer pushes on the github these tests are ran and it makes sure that it doesn'te break the exiting code.

and continuous deployment means like build the app after the code is merged into the main branch like `develop` and then upload it to play store and make the release.

So I saw the tests run and I see what's failling the current tests [bazel](https://github.com/oppia/oppia-android/actions/runs/3810553828/jobs/6485921621#step:17:308) [gradle](https://github.com/oppia/oppia-android/actions/runs/3810553830/jobs/6485906098#step:7:701)

I think maybe you missed some of the dimens value for high density devices like xxhdpi devices
image

**Comment by dmdbilal** [comment]
> @dmdbilal a workflow run is like running all the test cases and other ci/cd(continuous integration and deployment) on the GitHub to test for any errors.
> Continuous integration is like for the development whenever a developer pushes on the github these tests are ran and it makes sure that it doesn'te break the exiting code.
> 
> and continuous deployment means like build the app after the code is merged into the main branch like `develop` and then upload it to play store and make the release.
> 
> So I saw the tests run and I see what's failling the current tests [bazel](https://github.com/oppia/oppia-android/actions/runs/3810553828/jobs/6485921621#step:17:308) [gradle](https://github.com/oppia/oppia-android/actions/runs/3810553830/jobs/6485906098#step:7:701)
> 
> I think maybe you missed some of the dimens value for high density devices like xxhdpi devices
> image
> 

@yash10019coder I'll solve that issue and let you know soon.

**Comment by yash10019coder** [comment]
also merge with the latest develop.

**Comment by dmdbilal** [comment]
@yash10019coder I've added the sizes in all the dimens.xml.

**Comment by yash10019coder** [comment]
> @yash10019coder I've added the sizes in all the dimens.xml.

Ok @dmdbilal I'll rerun the tests workflow let's see if the tests are passing now if they we can merge your first PR.

**Comment by dmdbilal** [comment]
> > @yash10019coder I've added the sizes in all the dimens.xml.
> 
> Ok @dmdbilal I'll rerun the tests workflow let's see if the tests are passing now if they we can merge your first PR.

Ok 👍

**Comment by yash10019coder** [comment]
Hi @dmdbilal could you please make another PR as you've force-pushed into this PR. Remember to never do a force push in a PR as repositories where commits are pushed and are public should never be overwritten.

Thanks,
Yash

**Pull request closed without merging**